CatsApp/Core/Network/NetworkClient.swift

```diff
@@ -21,51 +21,50 @@ enum NetworkError: Error, CustomStringConvertible {
     }
 }
 
-protocol NetworkClientProtocol {
-    func request<T: Decodable>(_ endpoint: Endpoint) async throws -> T
+struct NetworkClient {
+    var requestData: (_ endpoint: Endpoint) async throws -> Data
```

Comment:
since we don't want that other change the requestData
```suggestion
    public internal(set) var requestData: (_ endpoint: Endpoint) async throws -> Data
```
this means that this closure will be public to read but internal to set. Internal is desirable because you update your implementation during ur testes by adding @testable import packages during your tests which allow u to access internal scope...

My explanation might not be super clear 😓 but if u want feel free to call me.